samples/zigbee/light_switch/src/main.c

```diff
@@ -0,0 +1,474 @@
+{
+	zb_ret_t zb_err_code;
+
+	if (bufid) {
+		LOG_INF("Bulb not found, try again");
+
+		zb_err_code = zigbee_schedule_alarm(find_light_bulb, bufid,
+						    MATCH_DESC_REQ_START_DELAY);
+		ZB_ERROR_CHECK(zb_err_code);
+		zb_err_code = zigbee_schedule_alarm(find_light_bulb_timeout, 0,
+							MATCH_DESC_REQ_TIMEOUT);
+		ZB_ERROR_CHECK(zb_err_code);
+	} else {
+		zb_err_code = zb_buf_get_out_delayed(find_light_bulb_timeout);
+		ZB_ERROR_CHECK(zb_err_code);
+	}
+}
+
+/**@brief Callback for detecting button press duration.
+ *
+ * @param[in]   button   BSP Button that was pressed.
+ */
+static void light_switch_button_handler(zb_uint8_t button)
+{
+	zb_ret_t zb_err_code;
+	zb_bool_t on_off;
+
+	if (dk_get_buttons() & button) {
+		atomic_set(&device_ctx.button.long_poll, ZB_TRUE);
+		on_off = (button == LIGHT_SWITCH_BUTTON_ON) ? ZB_TRUE : ZB_FALSE;
+
+		/* Allocate output buffer and send step command. */
+		zb_err_code = zb_buf_get_out_delayed_ext(light_switch_send_step,
+							 on_off, 0);
+		ZB_ERROR_CHECK(zb_err_code);
+
+		zb_err_code = zigbee_schedule_alarm(light_switch_button_handler,
+						  button, BUTTON_LONG_POLL_TMO);
+		if (zb_err_code == RET_OVERFLOW) {
+			LOG_WRN("Can not schedule another alarm, queue is full.");
+			atomic_set(&device_ctx.button.in_progress, ZB_FALSE);
+		} else {
+			ZB_ERROR_CHECK(zb_err_code);
+		}
+	} else {
+		atomic_set(&device_ctx.button.long_poll, ZB_FALSE);
+	}
+}
+
+/**@brief Zigbee stack event handler.
+ *
+ * @param[in]   bufid   Reference to the Zigbee stack buffer used to pass signal.
+ */
+void zboss_signal_handler(zb_bufid_t bufid)
+{
+	zb_zdo_app_signal_hdr_t   * p_sg_p = NULL;
+	zb_zdo_app_signal_type_t    sig    = zb_get_app_signal(bufid, &p_sg_p);
+	zb_ret_t                    status = ZB_GET_APP_SIGNAL_STATUS(bufid);
+	zb_ret_t                    zb_err_code;
+
+	/* Update network status LED */
+	zigbee_led_status_update(bufid, ZIGBEE_NETWORK_STATE_LED);
+
+	switch(sig) {
+	case ZB_BDB_SIGNAL_DEVICE_REBOOT:
+		/* fall-through */
+	case ZB_BDB_SIGNAL_STEERING:
+		/* Call default signal handler. */
+		ZB_ERROR_CHECK(zigbee_default_signal_handler(bufid));
+		if (status == RET_OK) {
+			/* Check the light device address */
+			if (device_ctx.bulb_params.short_addr == 0xFFFF) {
+				zb_err_code = zigbee_schedule_alarm(
+						    find_light_bulb, bufid,
+						    MATCH_DESC_REQ_START_DELAY);
+				ZB_ERROR_CHECK(zb_err_code);
+				zb_err_code = zigbee_schedule_alarm(
+						find_light_bulb_timeout, 0,
+						MATCH_DESC_REQ_TIMEOUT);
+				ZB_ERROR_CHECK(zb_err_code);
+				/* Do not free buffer - it will be reused by
+				 * find_light_bulb callback.
+				 */
+				bufid = 0;
+			}
+		}
+		break;
+	default:
+		/* Call default signal handler. */
+		ZB_ERROR_CHECK(zigbee_default_signal_handler(bufid));
+		break;
+	}
+
+	if (bufid) {
+		zb_buf_free(bufid);
+	}
+}
+
+/**@brief Function for application main entry.
+ */
+static void zigbee_app_thread(void)
```

Comment:
We've resolved for other samples to simly use `main` thread, instead of creating a new one, please have a look at the other PR.